### ISSUE TITLE
QueueBuildPolicy should take a CancellationToken

### ISSUE THREAD

**Issue by dmonroym** [dotnet/roslyn-tools#706]
To be consistent QueueBuildPolicy should take a CancellationToken
https://github.com/dotnet/roslyn-tools/blob/4ffedee20066f89c8be7a21d4580875f72a7d022/src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.cs#L372

**Comment by RikkiGibson** [comment]
Yeah it feels like any of the methods (particularly async methods) which call into the ADO client library methods that take CancellationToke, should pass down the cancellation token we have at the top level of the RIT.
